coralproject/talk
# [CORL-2875]: Improve consistency of Redis cache key for comment count

## What does this PR do?

This makes the comment count jsonp Redis cache keys more consistently use just `storyUrl`. This also updates so that both the comment count html and the comment count text html are saved under the same key, and then the `notext` data attribute is used by the client at injection time to determine which html elements are needed.

These changes mean that only one key will be saved for each comment count story url.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

none

## Does this PR introduce any new environment variables or feature flags?

no

## If any indexes were added, were they added to `INDEXES.md`?

n/a

## How do I test this PR?

You can test this PR by using a page that has comment counts included (such as multi-site test index and story pages). Set your `JSONP_RESPONSE_CACHE` to `true` and the `JSONP_CACHE_MAX_AGE` to something like `3m`. Start it up and visit the page. Then go into Redis (`docker exec -it redis redis-cli  `) and look at the keys (`KEYS *`). See that the keys for the comment counts for the stories you loaded should look like `rmc:CoralCount&url=STORYURL`. The comment counts should still work as expected. Test comment counts both with and without the `data-coral-notext` attribute set to `"true"`) and confirm that with that, it shows the `Comments`/`New` text, and without it it shows just a comment count.

## Where any tests migrated to React Testing Library?



## How do we deploy this PR?
